### ISSUE TITLE
AddPaymentSourcesToWallet "randomly" changing the default

### ISSUE THREAD

**Issue by adammathys** [solidusio/solidus#4004]
As of the introduction of #2913, we no longer set the default payment to one that was used on the current order. Instead, we implicitly pick the one with the biggest ID from all sources in a user's wallet. This can result in inconsistent behaviour where placing an order will change the default payment from the one the customer used and/or the one a customer has explicitly set as their default.

**Solidus Version**:
v2.11+

**To Reproduce**
1. Add two payment sources to a users wallet. Make the one with the smallest ID the default payment source.
2. Complete checkout using the default payment source.

**Current behavior**
After placing an order with the default payment source, the default will be changed to the wallet source with the largest ID. (Because `order.user.wallet_payment_sources.last` should implicitly order by ID.) Meaning we changed the user's default payment source without them ever selecting the new default.

**Expected behavior**
Based on previous behaviour, default payment source should be unchanged after placing the order.

**Comment by jarednorman** [comment]
Am I misunderstanding that the _previous_ behaviour was for the reusable source _on the order_ with the highest ID would previously have been selected as the default. That's how I'm reading [this](https://github.com/solidusio/solidus/pull/2913/files#diff-7fd6876e363aa5cb0d758438a2cf1523d888d90396ad36b1e25931855bb03358L31-L32).

**Comment by adammathys** [comment]
Correct, the previous behaviour would be to pick the source with the largest `Spree::Source#id` that was part of a valid payment on that order. 

The new logic instead picks the wallet source with the largest `Spree::WalletPaymentSource#id` from all of the sources in the wallet.
